xls/modules/zstd/memory/BUILD

```diff
@@ -0,0 +1,675 @@
+)
+
+xls_dslx_library(
+    name = "common_dslx",
+    srcs = [
+        "common.x",
+    ],
+    deps = [":axi_dslx"],
+)
+
+xls_dslx_test(
+    name = "common_dslx_test",
+    library = ":common_dslx",
+    tags = ["manual"],
+)
+
+CLOCK_PERIOD_PS = "750"
+
+common_codegen_args = {
+    "delay_model": "asap7",
+    "reset": "rst",
+    "worst_case_throughput": "1",
+    "use_system_verilog": "false",
+    "clock_period_ps": CLOCK_PERIOD_PS,
+    "clock_margin_percent": "20",
+    "multi_proc": "true",
+}
+
+xls_dslx_library(
+    name = "axi_reader_dslx",
+    srcs = ["axi_reader.x"],
+    deps = [
+        ":axi_dslx",
+        ":axi_st_dslx",
+        ":common_dslx",
+    ],
+)
+
+xls_dslx_test(
+    name = "axi_reader_dslx_test",
+    library = ":axi_reader_dslx",
+    tags = ["manual"],
+)
+
+axi_reader_codegen_args = common_codegen_args | {
+    "module_name": "axi_reader",
+    "pipeline_stages": "2",
+    "streaming_channel_data_suffix": "_data",
+    "flop_inputs_kind": "skid",
+    "flop_outputs_kind": "skid",
+    "clock_period_ps": "1800",
+}
+
+xls_dslx_verilog(
+    name = "axi_reader_verilog",
+    codegen_args = axi_reader_codegen_args,
+    dslx_top = "AxiReaderInst",
+    library = ":axi_reader_dslx",
+    opt_ir_args = {
+        "top": "__axi_reader__AxiReaderInst__AxiReader_0__16_32_4_4_4_3_2_14_next",
+    },
+    tags = ["manual"],
+    verilog_file = "axi_reader.v",
+)
+
+xls_benchmark_ir(
+    name = "axi_reader_opt_ir_benchmark",
+    src = ":axi_reader_verilog.opt.ir",
+    benchmark_ir_args = axi_reader_codegen_args | {
+        "pipeline_stages": "10",
+    },
+    tags = ["manual"],
+)
+
+verilog_library(
+    name = "axi_reader_verilog_lib",
+    srcs = [
+        ":axi_reader.v",
+    ],
+    tags = ["manual"],
+)
+
+synthesize_rtl(
+    name = "axi_reader_synth_asap7",
+    standard_cells = "@org_theopenroadproject_asap7sc7p5t_28//:asap7-sc7p5t_rev28_rvt",
+    tags = ["manual"],
+    top_module = "axi_reader",
+    deps = [
+        ":axi_reader_verilog_lib",
+    ],
+)
+
+benchmark_synth(
+    name = "axi_reader_benchmark_synth",
+    synth_target = ":axi_reader_synth_asap7",
+    tags = ["manual"],
+)
+
+place_and_route(
+    name = "axi_reader_place_and_route",
+    clock_period = CLOCK_PERIOD_PS,
+    core_padding_microns = 2,
+    min_pin_distance = "0.5",
+    placement_density = "0.30",
+    stop_after_step = "global_routing",
+    synthesized_rtl = ":axi_reader_synth_asap7",
+    tags = ["manual"],
+    target_die_utilization_percentage = "10",
+)
+
+xls_dslx_library(
+    name = "axi_stream_remove_empty_dslx",
+    srcs = ["axi_stream_remove_empty.x"],
+    deps = [
+        ":axi_st_dslx",
+    ],
+)
+
+xls_dslx_test(
+    name = "axi_stream_remove_empty_dslx_test",
+    library = ":axi_stream_remove_empty_dslx",
+    tags = ["manual"],
+)
+
+axi_stream_remove_empty_codegen_args = common_codegen_args | {
+    "module_name": "axi_stream_remove_empty",
+    "clock_period_ps": "1300",
```

Comment:
any reason the clock period is different here? should we keep a list of those at the top for easier tuning?

Reply:
The critical path delay from the IR benchmark for this proc exceeds the 750ps that we use throughout the ZSTD Decoder. In order to provide valid constraints for the scheduler we increased the clock period to the value that is close to the critical path delay. Moved this to a variable at the top of the file where the `750ps` clock period is also defined.